File: frontend/src/services/noteService.js
import api from './api';

export const noteService = {
  addNote: async (applicationId, content) => {
    const response = await api.post(`/applications/${applicationId}/notes`, { content });
    return response.data;
  },
  getNotes: async (applicationId) => {
    const response = await api.get(`/applications/${applicationId}/notes`);
    return response.data;
  },
  deleteNote: async (noteId) => {
    await api.delete(`/notes/${noteId}`);
  }
};

File: backend/src/main/java/com/jobpulse/service/NoteService.java
package com.jobpulse.service;

import com.jobpulse.dto.request.NoteRequest;
import com.jobpulse.dto.response.NoteResponse;

import java.util.List;

public interface NoteService {
    NoteResponse addNote(Long userId, Long applicationId, NoteRequest request);
    List<NoteResponse> getNotesByApplication(Long userId, Long applicationId);
    void deleteNote(Long userId, Long noteId);
}
